decentraland/screens/BuyLandScreen.js

```diff
@@ -53,16 +60,29 @@ export class BuyLandScreen extends React.Component {
 
     const onSuccess = () => {
       showInfo(`${typeDescription} bought successfully.`);
-      removeLandForSale(landForSale);
-      addToMyAssetsList(asset);
+    };
+
+    const onError = () => {
+      removeMyAssetFromList(asset);
+      addLandForSaleToList(landForSale);
     };
 
     showInfo(`Buying the ${typeDescription.toLowerCase()}...`);
-    _executeOrder(landForSale, account, onSuccess);
+
+    // Optimistic UI update
+    removeLandForSale(landForSale);
+    addToMyAssetsList(asset);
```

Comment:
Maybe in a future PR, but the reducer for this action should probably set a pending state that is then set to confirmed onSuccess

Reply:
Extracted to: https://github.com/tasitlabs/tasit/issues/253

Reply:
Okay, great - thanks. Resolving this.